image/image.go

```diff
@@ -205,11 +221,37 @@ func tryOCI(fileName string) (v1.Image, error) {
 		return nil, xerrors.New("no valid manifest")
 	}
 
-	// Support only first image
-	h := m.Manifests[0].Digest
-	img, err := index.Image(h)
-	if err != nil {
-		return nil, xerrors.New("invalid OCI image")
+	// Support image having tag separated by : , otherwise support first image
+
+	var img v1.Image
+
+	if inputTag != "" {
+		for _, manifest := range m.Manifests {
+			annotation := manifest.Annotations
+
+			tag := annotation[ispec.AnnotationRefName]
+			if tag == inputTag {
+				h := manifest.Digest
+
+				img, err = index.Image(h)
+				if err != nil {
+					return nil, xerrors.New("invalid OCI image")
+				}
+
+				break
+			}
+		}
+
+		if img == nil {
+			return nil, xerrors.New("invalid OCI image tag")
+		}
```

Comment:
Could you define a new function for these processes?

Reply:
Sure, I will define one